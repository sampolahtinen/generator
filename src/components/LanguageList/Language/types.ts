export interface LanguageProps {
  title: string;
  percentage: number;
}
