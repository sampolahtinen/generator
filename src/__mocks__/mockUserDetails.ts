export const mockUserDetails = {
  login: 'sampolahtinen',
  id: 19226286,
  node_id: 'MDQ6VXNlcjE5MjI2Mjg2',
  avatar_url: 'https://avatars2.githubusercontent.com/u/19226286?v=4',
  gravatar_id: '',
  url: 'https://api.github.com/users/sampolahtinen',
  html_url: 'https://github.com/sampolahtinen',
  followers_url: 'https://api.github.com/users/sampolahtinen/followers',
  following_url:
    'https://api.github.com/users/sampolahtinen/following{/other_user}',
  gists_url: 'https://api.github.com/users/sampolahtinen/gists{/gist_id}',
  starred_url:
    'https://api.github.com/users/sampolahtinen/starred{/owner}{/repo}',
  subscriptions_url: 'https://api.github.com/users/sampolahtinen/subscriptions',
  organizations_url: 'https://api.github.com/users/sampolahtinen/orgs',
  repos_url: 'https://api.github.com/users/sampolahtinen/repos',
  events_url: 'https://api.github.com/users/sampolahtinen/events{/privacy}',
  received_events_url:
    'https://api.github.com/users/sampolahtinen/received_events',
  type: 'User',
  site_admin: false,
  name: 'Sampo Lahtinen',
  company: 'plaj',
  blog: '',
  location: 'Berlin, Germany',
  email: 'test',
  hireable: true,
  bio:
    'A fellow interested in web app development with React, human-computer interaction and ux/ui design.',
  public_repos: 17,
  public_gists: 0,
  followers: 5,
  following: 4,
  created_at: '2016-05-06T14:37:35Z',
  updated_at: '2020-03-14T17:08:45Z',
  private_gists: 0,
  total_private_repos: 0,
  owned_private_repos: 0,
  disk_usage: 173622,
  collaborators: 0,
  two_factor_authentication: false,
  plan: {
    name: 'free',
    space: 976562499,
    collaborators: 0,
    private_repos: 10000
  }
};
