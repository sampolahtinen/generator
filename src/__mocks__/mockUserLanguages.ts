export const mockUserLanguages = {
  list: [
    { title: 'JavaScript', count: 5 },
    { title: 'CSS', count: 5 }
  ],
  totalCount: 10
};
