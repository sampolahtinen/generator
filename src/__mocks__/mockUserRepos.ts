export const mockUserRepos = [
  {
    id: 177132066,
    node_id: 'MDEwOlJlcG9zaXRvcnkxNzcxMzIwNjY=',
    name: 'connected-react-router',
    full_name: 'sampolahtinen/connected-react-router',
    private: false,
    owner: {
      login: 'sampolahtinen',
      id: 19226286,
      node_id: 'MDQ6VXNlcjE5MjI2Mjg2',
      avatar_url: 'https://avatars2.githubusercontent.com/u/19226286?v=4',
      gravatar_id: '',
      url: 'https://api.github.com/users/sampolahtinen',
      html_url: 'https://github.com/sampolahtinen',
      followers_url: 'https://api.github.com/users/sampolahtinen/followers',
      following_url:
        'https://api.github.com/users/sampolahtinen/following{/other_user}',
      gists_url: 'https://api.github.com/users/sampolahtinen/gists{/gist_id}',
      starred_url:
        'https://api.github.com/users/sampolahtinen/starred{/owner}{/repo}',
      subscriptions_url:
        'https://api.github.com/users/sampolahtinen/subscriptions',
      organizations_url: 'https://api.github.com/users/sampolahtinen/orgs',
      repos_url: 'https://api.github.com/users/sampolahtinen/repos',
      events_url: 'https://api.github.com/users/sampolahtinen/events{/privacy}',
      received_events_url:
        'https://api.github.com/users/sampolahtinen/received_events',
      type: 'User',
      site_admin: false
    },
    html_url: 'https://github.com/sampolahtinen/connected-react-router',
    description: 'A Redux binding for React Router v4',
    fork: true,
    url: 'https://api.github.com/repos/sampolahtinen/connected-react-router',
    forks_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/forks',
    keys_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/keys{/key_id}',
    collaborators_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/collaborators{/collaborator}',
    teams_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/teams',
    hooks_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/hooks',
    issue_events_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/issues/events{/number}',
    events_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/events',
    assignees_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/assignees{/user}',
    branches_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/branches{/branch}',
    tags_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/tags',
    blobs_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/git/blobs{/sha}',
    git_tags_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/git/tags{/sha}',
    git_refs_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/git/refs{/sha}',
    trees_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/git/trees{/sha}',
    statuses_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/statuses/{sha}',
    languages_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/languages',
    stargazers_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/stargazers',
    contributors_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/contributors',
    subscribers_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/subscribers',
    subscription_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/subscription',
    commits_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/commits{/sha}',
    git_commits_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/git/commits{/sha}',
    comments_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/comments{/number}',
    issue_comment_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/issues/comments{/number}',
    contents_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/contents/{+path}',
    compare_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/compare/{base}...{head}',
    merges_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/merges',
    archive_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/{archive_format}{/ref}',
    downloads_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/downloads',
    issues_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/issues{/number}',
    pulls_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/pulls{/number}',
    milestones_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/milestones{/number}',
    notifications_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/notifications{?since,all,participating}',
    labels_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/labels{/name}',
    releases_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/releases{/id}',
    deployments_url:
      'https://api.github.com/repos/sampolahtinen/connected-react-router/deployments',
    created_at: '2019-03-22T11:58:20Z',
    updated_at: '2019-03-22T11:58:22Z',
    pushed_at: '2019-03-18T20:30:38Z',
    git_url: 'git://github.com/sampolahtinen/connected-react-router.git',
    ssh_url: 'git@github.com:sampolahtinen/connected-react-router.git',
    clone_url: 'https://github.com/sampolahtinen/connected-react-router.git',
    svn_url: 'https://github.com/sampolahtinen/connected-react-router',
    homepage: '',
    size: 1067,
    stargazers_count: 0,
    watchers_count: 0,
    language: 'JavaScript',
    has_issues: false,
    has_projects: true,
    has_downloads: true,
    has_wiki: true,
    has_pages: false,
    forks_count: 0,
    mirror_url: 'test',
    archived: false,
    disabled: false,
    open_issues_count: 0,
    license: {
      key: 'mit',
      name: 'MIT License',
      spdx_id: 'MIT',
      url: 'https://api.github.com/licenses/mit',
      node_id: 'MDc6TGljZW5zZTEz'
    },
    forks: 0,
    open_issues: 0,
    watchers: 0,
    default_branch: 'master',
    permissions: 'test'
  }
];
