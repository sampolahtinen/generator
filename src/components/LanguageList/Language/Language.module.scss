@import '../../../styles/colors.scss';

.language {
  margin-bottom: 2rem;
  white-space: nowrap;

  .titleWrapper {
    margin-bottom: 1.2rem;
  }

  .percentage {
    color: $grey_300;
    margin-left: 0.8rem;
  }
}
