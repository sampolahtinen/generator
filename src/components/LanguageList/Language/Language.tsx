import React from 'react';
import styles from './Language.module.scss';
import BarGauge from '../../BarGauge';
import { LanguageProps } from './types';

const Language = ({ title, percentage }: LanguageProps) => {
  return (
    <div className={styles.language}>
      <div className={styles.titleWrapper}>
        <span className="highlighted-text">{title}</span>
        <span className={styles.percentage}>{percentage}%</span>
      </div>
      <BarGauge value={percentage} />
    </div>
  );
};

export default Language;
