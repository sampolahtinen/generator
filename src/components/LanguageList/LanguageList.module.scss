@import '../../styles/colors';
@import '../../styles/variables';

.languageList {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(150px, 1fr));
  grid-gap: 16px;
  list-style-type: none;
}
