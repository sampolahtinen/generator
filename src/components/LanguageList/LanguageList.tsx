import React from 'react';
import { UserLanguages, UserLanguageWithCount } from '../../store/user/types';
import Language from './Language/Language';
import styles from './LanguageList.module.scss';

export interface LanguageListProps {
  languages: UserLanguages;
}

const LanguageList = ({ languages }: LanguageListProps) => {
  return (
    <div className={styles.languageList}>
      {languages.list.map((lang: UserLanguageWithCount) => (
        <Language
          key={lang.title}
          title={lang.title}
          percentage={Math.ceil((lang.count / languages.totalCount) * 100)}
        />
      ))}
    </div>
  );
};

export default LanguageList;
