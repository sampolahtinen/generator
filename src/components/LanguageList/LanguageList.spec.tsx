import React from 'react';
import { shallow } from 'enzyme';
import LanguageList from './LanguageList';
import Language from './Language/Language';
import { mockUserLanguages } from '../../__mocks__/mockUserLanguages';

describe('LanguageList Component', () => {
  const wrapper = shallow(<LanguageList languages={mockUserLanguages} />);

  it('renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('displays languages', () => {
    expect(wrapper.find(Language)).toHaveLength(mockUserLanguages.list.length);
  });
});
