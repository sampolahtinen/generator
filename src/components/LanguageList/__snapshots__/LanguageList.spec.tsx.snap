// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageList Component renders 1`] = `
<div
  className="languageList"
>
  <Language
    key="JavaScript"
    percentage={50}
    title="JavaScript"
  />
  <Language
    key="CSS"
    percentage={50}
    title="CSS"
  />
</div>
`;
